crates/analyzer/src/db/queries/functions.rs

```diff
@@ -123,12 +136,6 @@ pub fn function_signature(
                                 arg.span,
                                 "instances of `Context` must be named `ctx`",
                             );
-                        } else if !function.parent(db).is_contract() {
```

Comment:
@cburgdorf This was a driveby change that should have been its own tiny PR; it's just an arbitrary restriction that shouldn't exist. There's no reason to prevent someone from writing a utility or library function that uses the context object.